src/biconnectivity/biconnect.jl

```diff
@@ -0,0 +1,70 @@
+"""
+Biconnections: A state type for Depth First Search that finds the biconnected components
+"""
+type Biconnections
+    low::Vector{Int}
+    depth::Vector{Int}
+    stack::Vector{Edge}
+    biconnected_comps::Vector{Vector{Edge}}
+    id::Int
+end
+
+function Biconnections(g::SimpleGraph)
+    n = nv(g)
+    return Biconnections(zeros(Int, n), zeros(Int, n), [], [], 0)
+end
+
+"""
+Computes the biconnected components of a graph `g`
+and returns a Vector of vectors containing each biconnected component
+"""
+function biconnected_components(g::SimpleGraph)
+    state = Biconnections(g)
+    for u in vertices(g)
+        state.depth[u] == 0 && visit!(g, state, u, u)
```

Comment:
LightGraph guidelines state that shortcuts like this are ok as long as state isn't being mutated. `visit!` is a mutator, so it's better - though more unwieldy - to put it in an `if` statement.